modules/core/src/matrix_expressions.cpp

```diff
@@ -1669,7 +1693,22 @@ MatExpr Mat::zeros(int rows, int cols, int type)
     return e;
 }
 
-MatExpr Mat::zeros(Size size, int type)
+#if defined(CV_COMPATIBLE_API)
+MatExpr Mat::zeros(int rows, int cols, int type)
```

Comment:
> matrix_expressions.cpp

Lets make compatibility parts as a header only (inline functions/methods). Only public headers.

Reply:
`MatExpr` is incomplete type, thus needs declaration & body separation.
In fact, I have already tried inline definition, and also changing the return type to `Mat` and statically casting the embedded return as well, without any success..
Accordingly, I separated functions returning `MatExpr` as an exception